engine-standalone-storage/src/sync/mod.rs

```diff
@@ -122,6 +125,45 @@ pub fn consume_message(storage: &mut crate::Storage, message: Message) -> Result
                     near_tx_hash
                 }
 
+                TransactionKind::FtTransferCall(args) => {
+                    // This transaction only has an impact on the aurora state
+                    // if the receiver is the Aurora contract
+                    if args.receiver_id.as_ref() == AURORA_ACCOUNT_ID {
+                        let mut connector = connector::EthConnectorContract::init_instance(io);
+                        let promise_args = connector.ft_transfer_call(
+                            env.predecessor_account_id.clone(),
+                            env.current_account_id.clone(),
+                            args,
+                            env.prepaid_gas,
+                        )?;
+                        let env = {
+                            let mut tmp = env.clone();
+                            // Since this would be executed as a callback, the predecessor_account_id
+                            // is now equal to the current_account_id
+                            tmp.predecessor_account_id = env.current_account_id;
+                            tmp
+                        };
```

Comment:
Fine for now, but could be optimised possible with a function that does exactly this and takes in the account_id as a parameter.

Reply:
So you would like to see it written as something like the following?

```rust
let env = env.clone().with_predecessor_account_id(env.current_account_id);
```